lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -156,16 +158,18 @@ struct FirMemory {
            numReadWritePorts == rhs.numReadWritePorts &&
            dataWidth == rhs.dataWidth && depth == rhs.depth &&
            readLatency == rhs.readLatency && writeLatency == rhs.writeLatency &&
-           readUnderWrite == rhs.readUnderWrite;
+           readUnderWrite == rhs.readUnderWrite &&
+           writeUnderWrite == rhs.writeUnderWrite;
   }
 };
 } // namespace
 
 static std::string getFirMemoryName(const FirMemory &mem) {
-  return llvm::formatv("FIRRTLMem_{0}_{1}_{2}_{3}_{4}_{5}_{6}_{7}",
+  return llvm::formatv("FIRRTLMem_{0}_{1}_{2}_{3}_{4}_{5}_{6}_{7}_{8}",
```

Comment:
Out of curiosity, are we generating the same thing that SFC does?  Is there a "correct" answer here?

Reply:
No, we're totally different. SFC generates logic for memories which were not blackboxed as inline logic in the module. There should be no issue in picking a convention for what we want to call the memory.

Reply:
This has been source of debate.  The standard is silent on many issues relating to memory.  The initial lowering stuck to the letter of the spec.  We unique memories, and don't add wrappers.

One of the problems is that SFC is lowering multiple times, in different ways, using heuristics.  I think we want to push the heuristics up in the pipeline and have the different lowerings just pick up appropriate tagged or typed memories.  This is pretty high on the lists of things to straighten out.

Reply:
The naming will now also include an optional `{9}` position in the name that includes a string representation of how the clocks are attached to write ports. If the first and third write ports are the same clock, but the second is different, you'll get `_aba`. If they're all the same, you'll get `aaa`.